src/main/java/org/testng/internal/Invoker.java

```diff
@@ -474,109 +387,101 @@ else if (! m_continueOnFailedConfiguration) {
 
     // check if there are failed @BeforeGroups
     String[] groups = method.getGroups();
-    if(null != groups && groups.length > 0) {
-      for(String group : groups) {
-        if(m_beforegroupsFailures.containsKey(group)) {
-          result = false;
-          break;
-        }
+    for (String group : groups) {
+      if (m_beforegroupsFailures.containsKey(group)) {
+        result = false;
+        break;
       }
     }
     return result;
   }
 
-   // Creates a token for tracking a unique invocation of a method on an instance.
-   // Is used when configFailurePolicy=continue.
-  private static Object getMethodInvocationToken(ITestNGMethod method, Object instance) {
-    return String.format("%s+%d+%d", instance.toString(), method.getCurrentInvocationCount(), method.getParameterInvocationCount());
+  private boolean isBeforeClassMethod(ITestNGMethod currentTestMethod) {
+    return currentTestMethod != null && m_methodInvocationResults.containsKey(currentTestMethod);
+  }
+
+  private Set<Object> getInvocationResults(IClass testClass) {
+    Class<?> cls = testClass.getRealClass();
+    Set<Object> set = null;
+    //We need to continuously search till either our Set is not null (or) till we reached
+    //Object class because it is very much possible that the test method is residing in a child class
+    //and maybe the parent method has configuration methods which may have had a failure
+    //So lets walk up the inheritance tree until either we find failures or till we
+    //reached the Object class.
+    while (true) {
+      if (cls.equals(Object.class)) {
+        break;
+      }
+      set = m_classInvocationResults.get(cls);
+      if (set != null) {
+        break;
```

Comment:
return set

Reply:
There is a return statement at the end of the method which takes care of returning the set.

Reply:
true, but you can make it easier to read

Reply:
@juherr - i am not sure i quite understand the concern here. I have merely extracted out the existing logic into a separate method, so that the method complexity is lowered. The current implementation also has sufficient comments that explains what the logic does. What more is expected ? It would be great if you could please clarify.

Reply:
The infinite loop is strange when you have break conditions.

Check https://github.com/cbeust/testng/pull/1582/files#r146972535